src/battle_arcade_game_board.c

```diff
@@ -0,0 +1,1442 @@
+static void BattleArcade_SetVramBanks(void);
+static void ov84_0223E9E4(BGCONFIG *config) ;
+static void ov84_0223EAAC(GAME_BOARD_WORK *work, u32 a1);
+static void ov84_0223EB08(GAME_BOARD_WORK *work, u32 a1);
+static void ov84_0223EB44(void);
+static void ov84_0223EB78(GAME_BOARD_WORK *work, u32 a1);
+static void BattleArcadeGameBoard_SetState(GAME_BOARD_WORK *work, int *state, int a2);
+static void ov84_0223EBE8(GAME_BOARD_WORK *work, int key);
+static void BattleArcadeGameBoard_SetCursorPos(GAME_BOARD_WORK *work, u8 cursorPos);
+static void BattleArcadeGameBoard_StartCursorSlowdown(GAME_BOARD_WORK *work);
+static u8 ov84_0223EC88(u8 area, u8 a2);
+static u16 BattleArcade_GetWinstreak(GAME_BOARD_WORK *work);
+static BOOL ov84_0223ECBC(GAME_BOARD_WORK *work, u16 type, u16 a2);
+static void ov84_0223ED00(GAME_BOARD_WORK *work, u16 type);
+static void ov84_0223ED6C(GAME_BOARD_WORK *work, u16 type, u16 a2);
+static u16 BattleArcade_Random(GAME_BOARD_WORK *work);
+static void ov84_0223EDF8(GAME_BOARD_WORK *work, u16 type);
+static void BattleArcadeGameBoard_StopCursor(GAME_BOARD_WORK *work, u8 type);
+static void ov84_0223EE74(GAME_BOARD_WORK *work);
+static void ov84_0223EFE0(GAME_BOARD_WORK *work);
+static void ov84_0223F094(GAME_BOARD_WORK *work);
+static u8 ov84_0223F178(GAME_BOARD_WORK*, u8);
+static void ov84_0223F1BC(GAME_BOARD_WORK *work);
+static BOOL BattleArcadeGameBoard_CheckButtonPress(GAME_BOARD_WORK *work);
+static void ov84_0223F28C(GAME_BOARD_WORK *work);
+static void ov84_0223F2B4(GAME_BOARD_SUB_3E8 *work, PARTY *playerParty, PARTY *opponentParty, u8 type);
+static Sprite *ov84_0223F374(GAME_BOARD_SUB_3E8 *work, u32 chara, u32 pal, u32 cell, u32 anim, u32 prio, int bgPrio, u8 display);
+static void ov84_0223F418(GAME_BOARD_SUB_3E8 *work);
+static void ov84_0223F480(void);
+static void ov84_0223F4B4(GAME_BOARD_SUB_3E8 *work);
+static void ov84_0223F538(GAME_BOARD_SUB_3E8 *work);
+static void ov84_0223F5E4(GAME_BOARD_SUB_3E8 *work, PARTY *playerParty, PARTY *opponentParty, u8 type);
+static void ov84_0223F714(GAME_BOARD_SUB_3E8 *work);
+static BATTLE_ARCADE_OBJECT *BattleArcadeObject_Create(GAME_BOARD_SUB_3E8 *work, u32 chara, u32 pal, u32 cell, u32 anim, u16 x, u16 y, u32 priority, int bgPrio, u8 display);
+static void *BattleArcadeObj_Delete(BATTLE_ARCADE_OBJECT *obj);
+static void BattleArcadeObj_SetVisible(BATTLE_ARCADE_OBJECT *obj, int flag);
+static void BattleArcadeObj_SetPos(BATTLE_ARCADE_OBJECT *obj, u16 x, u16 y);
+static void BattleArcadeObj_SetAnimation(BATTLE_ARCADE_OBJECT *obj, u32 val);
+static void ov84_0223F894(BATTLE_ARCADE_OBJECT *obj, POKEMON *mon);
+static void ov84_0223F8A8(BATTLE_ARCADE_OBJECT *obj, int flag);
+static void ov84_0223F8B4(BGCONFIG *config, WINDOW *window);
+static void ov84_0223F8E4(WINDOW *window);
+
+BOOL BattleArcadeGameBoard_InitOverlay(OVY_MANAGER *man, int *state) {
+    int i;
+    GAME_BOARD_WORK *work;
+    GAME_BOARD_ARGS *args;
+    
+    HandleLoadOverlay(FS_OVERLAY_ID(OVY_80), 2);
```

Comment:
nit: rename `OVY_80` -> `battle_arcade_board` or something similar

Reply:
OVY_80 is the frontier overlay iirc, not entirely sure so I held off on naming it for now